packages/core/src/hosted-form/iframe-content/hosted-input-events.ts

```diff
@@ -53,12 +57,25 @@ export type HostedInputEvent =
     | HostedInputFocusEvent
     | HostedInputSubmitSuccessEvent
     | HostedInputSubmitErrorEvent
-    | HostedInputValidateEvent;
+    | HostedInputValidateEvent
+    | HostedInputVaultingSucceededEvent
+    | HostedInputVaultingErrorEvent;
 
 export interface HostedInputAttachSuccessEvent {
     type: HostedInputEventType.AttachSucceeded;
 }
 
+export interface HostedInputVaultingSucceededEvent {
+    type: HostedInputEventType.VaultingSucceeded;
+}
+
+export interface HostedInputVaultingErrorEvent {
+    type: HostedInputEventType.VaultingFailed;
+    payload?: {
+        errors?: string[];
```

Comment:
For payment submission, we return a rich error object to the parent window as the submission could potentially trigger human verification. I assume for vaulting, we need a similar form of protection to prevent carding attack. If that's the case, `string[]` might not be the most suitable interface. 

PS: I can't see where the vaulting submission is triggered so I can only assume we have or plan to have reCaptcha verification.

Reply:
@davidchin yeah, it would be done in the future